python/pyarrow/tests/test_extension_type.py

```diff
@@ -1127,3 +1134,12 @@ def test_cpp_extension_in_python(tmpdir):
     reconstructed_array = batch.column(0)
     assert reconstructed_array.type == uuid_type
     assert reconstructed_array == array
+
+    storage = pa.array([b'0onmlkjihgfedcba']*4, pa.binary(16))
+    ext_array = pa.ExtensionArray.from_storage(uuid_type, storage)
+    assert isinstance(ext_array, array_cls)
```

Comment:
From memory, before to this change, `pa.ExtensionArray.from_storage` would have returned a `pyarrow.ExtensionArray`.